MainEvent/main.cpp

```diff
@@ -94,6 +103,51 @@ void MoveRobot() {
   //Roboteq.sendAngles();
 }
 
+bool sortingMethod (Mat& a, Mat& b){
+    return a.Y > b.Y;
+}
+
+void matchingMethod(){
+  //so these are sorted right now
+  sort(ballCoordR.begin(), ballCoordR.end(), sortingMethod());
+  sort(ballCoordL.begin(), ballCoordL.end(), sortingMethod());
+
+  stack<Mat> correctedR;
+  stack<Mat> correctedL;
+
+  for (auto rball: ballCoordR){
+    if (rball.X < rightBound){
+      correctedR.push(rball);
+    }
+  }
+  for (auto lball: ballCoordL){
+    if (lball.X > leftBound){
+      correctedL.push(lball);
+    }
+  }
+
+  while (!correctedR.empty()){
+    Mat rball = correctedR.top();
+    while (true){
+      if (correctedL.empty()) break;
+      Mat ball = correctedL.top();
+
+      if (abs(rball.Y - ball.Y) < marginOfError){
+        Mat lball = correctedL.pop();
+        pair<Mat,Mat> newPair;
+        newPair.first = lball;
+        newPair.second = rball;
+        pairedBalls.push_back(newPair);
+        break;
+      }
+      //rball.Y > ball.Y means that ball.Y wont fit with any rball
+      else if(rball.Y > ball.Y) correctL.pop();
```

Comment:
I think that's a typo. Did you mean `correctR.pop();`?

Reply:
No, you want to pop correctedL here, because if the lball.Y from correctedL is smaller than the current rball, that lball wont satisfy any future rballs

Reply:
there was a typo though, I wrote correctL instead of correctedL